openedx/core/djangoapps/course_groups/views.py

```diff
@@ -384,16 +388,26 @@ def remove_user_from_cohort(request, course_key_string, cohort_id):
         return json_http_response({'success': False,
                                    'msg': 'No username specified'})
 
-    cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
     try:
         user = User.objects.get(username=username)
-        cohort.users.remove(user)
-        return json_http_response({'success': True})
     except User.DoesNotExist:
         log.debug('no user')
         return json_http_response({'success': False,
                                    'msg': "No user '{0}'".format(username)})
 
+    try:
+        membership = CohortMembership.objects.get(user=user, course_id=course_key)
+
+        # This outer_atomic() wrapper ensures that the CohortMembership and course_user_group models update atomically.
+        # This will be commited to the database immediately.
+        with outer_atomic():
+            membership.delete()
```

Comment:
How is this different than just allowing an automatic `transaction.atomic` to wrap this view?